x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/IndicesPermission.java

```diff
@@ -40,34 +44,30 @@
  * A permission that is based on privileges for index related actions executed
  * on specific indices
  */
-public final class IndicesPermission implements Iterable<IndicesPermission.Group> {
+public final class IndicesPermission {
 
     public static final IndicesPermission NONE = new IndicesPermission();
+    private static final Automaton systemIndicesAutomaton = Automatons.patterns(SystemIndicesNames.indexNames());
```

Comment:
Just to check we're fine with building Automatons at class initialization.

Reply:
I am fine with it since this is not going to be a big one